### ISSUE TITLE
Czech restaurant (and possibly DART) BLEU seems suspiciously low

### ISSUE THREAD

**Issue by tuetschek** [GEM-benchmark/GEM-metrics#27]
BLEU of 2-3% is very low – maybe there's a problem with the references and/or the outputs (wrong order etc.)?

**Comment by jordiclive** [comment]
@tuetschek Yes, the BLEU for DART should be >45%, so it might be a good idea to update the v1 paper as it is a bug. 

I've been looking at DART, E2E Clean, and WEBNLG+ 2020 as am familiar with them. I get close to zero scores when using a general submission. Not sure the exact issue, as I pass in the keys.

One thing I noticed was for the E2E Clean test and validation set, there are identical inputs, so there are 4693 examples, rather than 1847 with multiple references, which would affect the metric calculation.

**Comment by tuetschek** [comment]
@jordiclive That's weird... yeah 4693 sounds like the multiple references are ignored? Can you please send me the submission file you're trying this with?

**Comment by jordiclive** [comment]
I think for E2E its in the datasets loading script.  As when you look at it with datasets.load_dataset('GEM/e2e_nlg') it is like that for validation and testing. So we'd have to edit https://huggingface.co/datasets/GEM/e2e_nlg/blob/main/e2e_nlg.py. 

For the submission script, I would be very interested why it doesn't work for these datasets. Let me get back to you with the general submission script and with separate reference files where it works.

**Comment by tuetschek** [comment]
@jordiclive I'm sorry for non-responsiveness, I got piled under teaching and admin, now I'm trying to catch up a little. Are you still able to have a look at the HF loader for E2E? It would be great if you could fix it! I'll finally have a look at your email with the submission script.

**Comment by tuetschek** [comment]
@jordiclive OK it looks like the problem is (compatibility?) different for each set:
* For DART, it looks like GEM-metrics is currently set *not* to compute scores on DART at all. The corresponding line in the config is commented out. I don't know the reason for this unfortunately.
* For E2E, it's exactly as you say – everything is single-ref and needs fixing, preferrably at Huggingface.
* For WebNLG, the problem seems to be compatibility with Huggingface. GEM-metrics is set to load the references from Huggingface, and in the HF version, the WebNLG references file has `target` and `references` keys for each instance. The `target` key is single-ref, it's just the first element of `references`. GEM-metrics always just looks for `target`, `references` is an unknown key for it.

I'm not exactly sure how to solve DART & WebNLG – @sebastianGehrmann , do you know why DART is now not used at all? And what do you propose to do with WebNLG – do we start looking at the `references` key? Or do we fix it on HF as well? I don't see the point of having `target` and `references` in this way, it's just confusing and you should always want to use multi-ref 🤔.

**Comment by jordiclive** [comment]
@tuetschek No worries. Thanks for investigating! lol quite coincidental. 

For the original GEM paper, the DART results were slightly above zero which I remember thinking looked strange when the paper came out. But that would be a different issue as something was computed.

It sounds like changing it on HF side for WebNLG would be cleanest. The only problem is I think for BigScience i wrote prompts using the references key for GEM WebNLG and we are currently running evaluation.

I will have a stab at the E2E loading script.

**Comment by jordiclive** [comment]
@tuetschek I changed the E2E loading script [here](https://huggingface.co/datasets/GEM/e2e_nlg/tree/e2e_fix) to use multi-references. @sebastianGehrmann can we merge this?

I investigated the `target` and `references:`
For each loading script, you have to define the types of the dataset 'features' for all splits (there could be a way to define separate feature types for each subset). So that is probably why `target` and` references` were made, as for training you want target to be `[str]` and references `List[str]`. I actually see this in several of the GEM datasets. For training, the references are not to be used (empty lists) and for testing and validation, the target is not meant to be used.

I followed this format for the E2E. Therefore, I think we should write some code to start looking at the references key for GEM? as otherwise, we would have to change lots of datasets on HF side and affect backward compat.

**Comment by jordiclive** [comment]
As for DART, are you able to run it uncommented and check if it marries up with my manual result? As not computing a score sounds different to the original issue of scores being extremely low.

**Comment by tuetschek** [comment]
@jordiclive I can see a "New Pull Request" button on the "[Community" tab](https://huggingface.co/datasets/GEM/references/discussions) for both – is that the place? Not sure how you can transform your branch into the PR, but I guess it's possible to do using the [command-line PR workflow](https://huggingface.co/docs/hub/repositories-pull-requests-discussions#how-do-i-manage-pull-requests-locally)?

**Comment by jordiclive** [comment]
Thanks! was struggling to change one file in references without downloading every single dataset. Hopefully can close this issue, when https://huggingface.co/datasets/GEM/references/discussions/1 is resolved.

**Comment by tuetschek** [comment]
@jordiclive That's great, thanks! I guess I'll still have to have a look at Czech restaurant, but I hope to do that in the next few days 🙂

**Comment by tuetschek** [comment]
OK so far it looks like CS Restaurants has also problems with some files, so I opened another issue there (when the files are OK, the scores seem fine now).
